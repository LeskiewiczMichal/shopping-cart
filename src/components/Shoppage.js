import ItemCard from './ItemCard';
import '../styles/shoppage.css';

const Shoppage = (props) => {
  const items = ['CD player', 'Mystery box', 'Book', 'Xbox', 'Bed', 'Plant'];

  const itemCards = items.map((item) => {
    return <ItemCard addToCart={props.addToCart} name={item} />;
  });

  return <div className="shoppage--wrapper">{itemCards}</div>;
};

export default Shoppage;
