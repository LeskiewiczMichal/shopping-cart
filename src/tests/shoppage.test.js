import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Shoppage from '../components/Shoppage';

test('Shop page snapshot', () => {
  const { container } = render(<Shoppage />);

  expect(container).toMatchSnapshot();
});
