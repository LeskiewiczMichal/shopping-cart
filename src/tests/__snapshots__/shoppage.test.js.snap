// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop page snapshot 1`] = `
<div>
  <div
    class="shoppage--wrapper"
  >
    <div
      class="itemcard--wrapper"
    >
      <span
        class="itemcard--name"
      >
        Name: 
        <p>
          CD player
        </p>
      </span>
      <label
        class="itemcard--quantity"
      >
        Quantity:
        <input
          id="quantity"
          min="1"
          name="quantity"
          type="number"
          value="1"
        />
      </label>
      <button
        class="itemcard--button"
      >
        Add to cart
      </button>
    </div>
    <div
      class="itemcard--wrapper"
    >
      <span
        class="itemcard--name"
      >
        Name: 
        <p>
          Mystery box
        </p>
      </span>
      <label
        class="itemcard--quantity"
      >
        Quantity:
        <input
          id="quantity"
          min="1"
          name="quantity"
          type="number"
          value="1"
        />
      </label>
      <button
        class="itemcard--button"
      >
        Add to cart
      </button>
    </div>
    <div
      class="itemcard--wrapper"
    >
      <span
        class="itemcard--name"
      >
        Name: 
        <p>
          Book
        </p>
      </span>
      <label
        class="itemcard--quantity"
      >
        Quantity:
        <input
          id="quantity"
          min="1"
          name="quantity"
          type="number"
          value="1"
        />
      </label>
      <button
        class="itemcard--button"
      >
        Add to cart
      </button>
    </div>
    <div
      class="itemcard--wrapper"
    >
      <span
        class="itemcard--name"
      >
        Name: 
        <p>
          Xbox
        </p>
      </span>
      <label
        class="itemcard--quantity"
      >
        Quantity:
        <input
          id="quantity"
          min="1"
          name="quantity"
          type="number"
          value="1"
        />
      </label>
      <button
        class="itemcard--button"
      >
        Add to cart
      </button>
    </div>
    <div
      class="itemcard--wrapper"
    >
      <span
        class="itemcard--name"
      >
        Name: 
        <p>
          Bed
        </p>
      </span>
      <label
        class="itemcard--quantity"
      >
        Quantity:
        <input
          id="quantity"
          min="1"
          name="quantity"
          type="number"
          value="1"
        />
      </label>
      <button
        class="itemcard--button"
      >
        Add to cart
      </button>
    </div>
    <div
      class="itemcard--wrapper"
    >
      <span
        class="itemcard--name"
      >
        Name: 
        <p>
          Plant
        </p>
      </span>
      <label
        class="itemcard--quantity"
      >
        Quantity:
        <input
          id="quantity"
          min="1"
          name="quantity"
          type="number"
          value="1"
        />
      </label>
      <button
        class="itemcard--button"
      >
        Add to cart
      </button>
    </div>
  </div>
</div>
`;
